### ISSUE TITLE
[Grouped products] Add searching support by product name AND/OR sku to the /products rest endpoint

### ISSUE THREAD

**Issue by mdperez86** [woothemes/woocommerce#41662]
**User story**

- As a merchant,
- I want to sell a set of similar products
- So that I can earn more money on each order

**Problem**

The [current endpoint implementation](https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-products) does not support searching product by `name` or `sku` at the same time. It has a `sku` only search but it does not work if the `search` param is sent too.

**Design**

AkVNGImLgSqCObTQ3idVn7-fi-3427_365209

**Acceptance criteria**

- [ ] The user can search the list by product name or SKU.

Depends on https://github.com/woocommerce/woocommerce/issues/35146

**Comment by pmcpinto** [comment]
@mdperez86 is this still blocked?

**Comment by mdperez86** [comment]
> @mdperez86 is this still blocked?

Yep, I will ping this issue as part of the rest API feedback thread so we can have a more precise discussion there.

**Comment by pmcpinto** [comment]
Considering the feedback shared in the [GH discussion](https://github.com/woocommerce/woocommerce/discussions/44199#discussioncomment-8534958), I think we should focus on searching by Name. And implement the search by SKU at a later stage. Maikel, do you agree?
cc @louwie17 @j111q in case you have thoughts on this.

**Comment by mdperez86** [comment]
> Considering the feedback shared in the [GH discussion](https://github.com/woocommerce/woocommerce/discussions/44199#discussioncomment-8534958), I think we should focus on searching by Name. And implement the search by SKU at a later stage. Maikel, do you agree? cc @louwie17 @j111q in case you have thoughts on this.

Yep, I do agree, but that is the purpose of this issue since searching by product name only is already implemented.

**Comment by pmcpinto** [comment]
I'm going to close this one. It refers to the new editor that will be removed.
